Matlab wrapper not compiling

Hi, I'm want to install GTSAM Matlab toolbox. Using Matlab 2023a, Ubuntu 22.04 LTS. ALways I try to "make" inside appears the following errors:
MatlabWrapper] Ignoring classes: ['gtsam::Point2', 'gtsam::Point3', 'gtsam::CustomFactor']
[100%] Building CXX object matlab/CMakeFiles/gtsam_unstable_matlab_wrapper.dir/__/wrap/gtsam_unstable/gtsam_unstable_wrapper.cpp.o
/home/danielsofia/Downloads/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp: In function ‘void gtsamCal3_S2Stereo_vector_1468(int, mxArray**, int, const mxArray**)’:
/home/danielsofia/Downloads/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:18824:45: error: ‘Vector6’ was not declared in this scope; did you mean ‘gtsam::Vector6’?
18824 |   out[0] = wrap_shared_ptr(std::make_shared<Vector6>(obj->vector()),"Vector6", false);
      |                                             ^~~~~~~
      |                                             gtsam::Vector6
In file included from /home/danielsofia/Downloads/gtsam/wrap/matlab.h:25,
                 from /home/danielsofia/Downloads/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/home/danielsofia/Downloads/gtsam/gtsam/base/Vector.h:51:9: note: ‘gtsam::Vector6’ declared here
   51 |   using Vector##N = Eigen::Matrix<double, N, 1>; \
      |         ^~~~~~
/home/danielsofia/Downloads/gtsam/gtsam/base/Vector.h:56:1: note: in expansion of macro ‘GTSAM_MAKE_VECTOR_DEFS’
   56 | GTSAM_MAKE_VECTOR_DEFS(6)
      | ^~~~~~~~~~~~~~~~~~~~~~
/home/danielsofia/Downloads/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:18824:53: error: no matching function for call to ‘make_shared<<expression error> >(gtsam::Vector6)’
18824 |   out[0] = wrap_shared_ptr(std::make_shared<Vector6>(obj->vector()),"Vector6", false);
      |                            ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
In file included from /usr/include/c++/11/memory:77,
                 from /home/danielsofia/Downloads/gtsam/gtsam/base/Testable.h:40,
                 from /home/danielsofia/Downloads/gtsam/gtsam/base/Manifold.h:23,
                 from /home/danielsofia/Downloads/gtsam/gtsam/base/Lie.h:26,
                 from /home/danielsofia/Downloads/gtsam/gtsam/base/VectorSpace.h:11,
                 from /home/danielsofia/Downloads/gtsam/gtsam/geometry/Point2.h:20,
                 from /home/danielsofia/Downloads/gtsam/wrap/matlab.h:27,
                 from /home/danielsofia/Downloads/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/bits/shared_ptr.h:875:5: note: candidate: ‘template<class _Tp, class ... _Args> std::shared_ptr<_Tp> std::make_shared(_Args&& ...)’
  875 |     make_shared(_Args&&... __args)
      |     ^~~~~~~~~~~
/usr/include/c++/11/bits/shared_ptr.h:875:5: note:   template argument deduction/substitution failed:
/home/danielsofia/Downloads/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:18824:53: error: template argument 1 is invalid
18824 |   out[0] = wrap_shared_ptr(std::make_shared<Vector6>(obj->vector()),"Vector6", false);
      |                            ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
In file included from /usr/include/x86_64-linux-gnu/c++/11/bits/c++allocator.h:33,
                 from /usr/include/c++/11/bits/allocator.h:46,
                 from /usr/include/c++/11/string:41,
                 from /home/danielsofia/Downloads/gtsam/gtsam/base/types.h:28,
                 from /home/danielsofia/Downloads/gtsam/gtsam/global_includes.h:22,
                 from /home/danielsofia/Downloads/gtsam/gtsam/base/Vector.h:28,
                 from /home/danielsofia/Downloads/gtsam/wrap/matlab.h:25,
                 from /home/danielsofia/Downloads/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/ext/new_allocator.h: In instantiation of ‘void __gnu_cxx::new_allocator<_Tp>::construct(_Up*, _Args&& ...) [with _Up = gtsam::GaussianConditional; _Args = {std::shared_ptr<gtsam::GaussianConditional>&}; _Tp = gtsam::GaussianConditional]’:
/usr/include/c++/11/bits/alloc_traits.h:516:17:   required from ‘static void std::allocator_traits<std::allocator<_CharT> >::construct(std::allocator_traits<std::allocator<_CharT> >::allocator_type&, _Up*, _Args&& ...) [with _Up = gtsam::GaussianConditional; _Args = {std::shared_ptr<gtsam::GaussianConditional>&}; _Tp = gtsam::GaussianConditional; std::allocator_traits<std::allocator<_CharT> >::allocator_type = std::allocator<gtsam::GaussianConditional>]’
/usr/include/c++/11/bits/shared_ptr_base.h:519:39:   required from ‘std::_Sp_counted_ptr_inplace<_Tp, _Alloc, _Lp>::_Sp_counted_ptr_inplace(_Alloc, _Args&& ...) [with _Args = {std::shared_ptr<gtsam::GaussianConditional>&}; _Tp = gtsam::GaussianConditional; _Alloc = std::allocator<gtsam::GaussianConditional>; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]’
/usr/include/c++/11/bits/shared_ptr_base.h:650:16:   required from ‘std::__shared_count<_Lp>::__shared_count(_Tp*&, std::_Sp_alloc_shared_tag<_Alloc>, _Args&& ...) [with _Tp = gtsam::GaussianConditional; _Alloc = std::allocator<gtsam::GaussianConditional>; _Args = {std::shared_ptr<gtsam::GaussianConditional>&}; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]’
/usr/include/c++/11/bits/shared_ptr_base.h:1342:14:   required from ‘std::__shared_ptr<_Tp, _Lp>::__shared_ptr(std::_Sp_alloc_shared_tag<_Tp>, _Args&& ...) [with _Alloc = std::allocator<gtsam::GaussianConditional>; _Args = {std::shared_ptr<gtsam::GaussianConditional>&}; _Tp = gtsam::GaussianConditional; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]’
/usr/include/c++/11/bits/shared_ptr.h:409:59:   required from ‘std::shared_ptr<_Tp>::shared_ptr(std::_Sp_alloc_shared_tag<_Tp>, _Args&& ...) [with _Alloc = std::allocator<gtsam::GaussianConditional>; _Args = {std::shared_ptr<gtsam::GaussianConditional>&}; _Tp = gtsam::GaussianConditional]’
/usr/include/c++/11/bits/shared_ptr.h:862:14:   required from ‘std::shared_ptr<_Tp> std::allocate_shared(const _Alloc&, _Args&& ...) [with _Tp = gtsam::GaussianConditional; _Alloc = std::allocator<gtsam::GaussianConditional>; _Args = {std::shared_ptr<gtsam::GaussianConditional>&}]’
/usr/include/c++/11/bits/shared_ptr.h:878:39:   required from ‘std::shared_ptr<_Tp> std::make_shared(_Args&& ...) [with _Tp = gtsam::GaussianConditional; _Args = {std::shared_ptr<gtsam::GaussianConditional>&}]’
/home/danielsofia/Downloads/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:28797:72:   required from here
/usr/include/c++/11/ext/new_allocator.h:162:11: error: no matching function for call to ‘gtsam::GaussianConditional::GaussianConditional(std::shared_ptr<gtsam::GaussianConditional>&)’
  162 |         { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/danielsofia/Downloads/gtsam/gtsam/linear/GaussianBayesNet.h:23,
                 from /home/danielsofia/Downloads/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:134:
/home/danielsofia/Downloads/gtsam/gtsam/linear/GaussianConditional.h:83:5: note: candidate: ‘template<class KEYS> gtsam::GaussianConditional::GaussianConditional(const KEYS&, size_t, const gtsam::VerticalBlockMatrix&, const SharedDiagonal&)’
   83 |     GaussianConditional(
      |     ^~~~~~~~~~~~~~~~~~~
/home/danielsofia/Downloads/gtsam/gtsam/linear/GaussianConditional.h:83:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/11/bits/c++allocator.h:33,
                 from /usr/include/c++/11/bits/allocator.h:46,
                 from /usr/include/c++/11/string:41,
                 from /home/danielsofia/Downloads/gtsam/gtsam/base/types.h:28,
                 from /home/danielsofia/Downloads/gtsam/gtsam/global_includes.h:22,
                 from /home/danielsofia/Downloads/gtsam/gtsam/base/Vector.h:28,
                 from /home/danielsofia/Downloads/gtsam/wrap/matlab.h:25,
                 from /home/danielsofia/Downloads/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/ext/new_allocator.h:162:11: note:   candidate expects 4 arguments, 1 provided
  162 |         { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/danielsofia/Downloads/gtsam/gtsam/linear/GaussianBayesNet.h:23,
                 from /home/danielsofia/Downloads/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:134:
/home/danielsofia/Downloads/gtsam/gtsam/linear/GaussianConditional.h:74:5: note: candidate: ‘template<class TERMS> gtsam::GaussianConditional::GaussianConditional(const TERMS&, size_t, const Vector&, const SharedDiagonal&)’
   74 |     GaussianConditional(const TERMS& terms,
      |     ^~~~~~~~~~~~~~~~~~~
/home/danielsofia/Downloads/gtsam/gtsam/linear/GaussianConditional.h:74:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/11/bits/c++allocator.h:33,
                 from /usr/include/c++/11/bits/allocator.h:46,
                 from /usr/include/c++/11/string:41,
                 from /home/danielsofia/Downloads/gtsam/gtsam/base/types.h:28,
                 from /home/danielsofia/Downloads/gtsam/gtsam/global_includes.h:22,
                 from /home/danielsofia/Downloads/gtsam/gtsam/base/Vector.h:28,
                 from /home/danielsofia/Downloads/gtsam/wrap/matlab.h:25,
                 from /home/danielsofia/Downloads/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/ext/new_allocator.h:162:11: note:   candidate expects 4 arguments, 1 provided
  162 |         { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/danielsofia/Downloads/gtsam/gtsam/linear/GaussianBayesNet.h:23,
                 from /home/danielsofia/Downloads/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:134:
/home/danielsofia/Downloads/gtsam/gtsam/linear/GaussianConditional.h:66:5: note: candidate: ‘gtsam::GaussianConditional::GaussianConditional(gtsam::Key, const Vector&, const Matrix&, gtsam::Key, const Matrix&, gtsam::Key, const Matrix&, const SharedDiagonal&)’
   66 |     GaussianConditional(Key key, const Vector& d, const Matrix& R, Key parent1,
      |     ^~~~~~~~~~~~~~~~~~~
/home/danielsofia/Downloads/gtsam/gtsam/linear/GaussianConditional.h:66:5: note:   candidate expects 8 arguments, 1 provided
/home/danielsofia/Downloads/gtsam/gtsam/linear/GaussianConditional.h:61:5: note: candidate: ‘gtsam::GaussianConditional::GaussianConditional(gtsam::Key, const Vector&, const Matrix&, gtsam::Key, const Matrix&, const SharedDiagonal&)’
   61 |     GaussianConditional(Key key, const Vector& d, const Matrix& R, Key parent1,
      |     ^~~~~~~~~~~~~~~~~~~
/home/danielsofia/Downloads/gtsam/gtsam/linear/GaussianConditional.h:61:5: note:   candidate expects 6 arguments, 1 provided
/home/danielsofia/Downloads/gtsam/gtsam/linear/GaussianConditional.h:57:5: note: candidate: ‘gtsam::GaussianConditional::GaussianConditional(gtsam::Key, const Vector&, const Matrix&, const SharedDiagonal&)’
   57 |     GaussianConditional(Key key, const Vector& d, const Matrix& R,
      |     ^~~~~~~~~~~~~~~~~~~
/home/danielsofia/Downloads/gtsam/gtsam/linear/GaussianConditional.h:57:5: note:   candidate expects 4 arguments, 1 provided
/home/danielsofia/Downloads/gtsam/gtsam/linear/GaussianConditional.h:54:5: note: candidate: ‘gtsam::GaussianConditional::GaussianConditional()’
   54 |     GaussianConditional() {}
      |     ^~~~~~~~~~~~~~~~~~~
/home/danielsofia/Downloads/gtsam/gtsam/linear/GaussianConditional.h:54:5: note:   candidate expects 0 arguments, 1 provided
/home/danielsofia/Downloads/gtsam/gtsam/linear/GaussianConditional.h:40:22: note: candidate: ‘gtsam::GaussianConditional::GaussianConditional(const gtsam::GaussianConditional&)’
   40 |   class GTSAM_EXPORT GaussianConditional :
      |                      ^~~~~~~~~~~~~~~~~~~
/home/danielsofia/Downloads/gtsam/gtsam/linear/GaussianConditional.h:40:22: note:   no known conversion for argument 1 from ‘std::shared_ptr<gtsam::GaussianConditional>’ to ‘const gtsam::GaussianConditional&’
/home/danielsofia/Downloads/gtsam/gtsam/linear/GaussianConditional.h:40:22: note: candidate: ‘gtsam::GaussianConditional::GaussianConditional(gtsam::GaussianConditional&&)’
/home/danielsofia/Downloads/gtsam/gtsam/linear/GaussianConditional.h:40:22: note:   no known conversion for argument 1 from ‘std::shared_ptr<gtsam::GaussianConditional>’ to ‘gtsam::GaussianConditional&&’
[100%] Linking CXX shared module ../wrap/gtsam_unstable_mex/gtsam_unstable_wrapper.mexa64
[100%] Built target gtsam_unstable_matlab_wrapper
make[2]: *** [matlab/CMakeFiles/gtsam_matlab_wrapper.dir/build.make:96: matlab/CMakeFiles/gtsam_matlab_wrapper.dir/__/wrap/gtsam/gtsam_wrapper.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:33375: matlab/CMakeFiles/gtsam_matlab_wrapper.dir/all] Error 2
make: *** [Makefile:166: all] Error 2



And all finished.

Can you help me please?  Thanks a lot

--- a/gtsam/geometry/geometry.i
+++ b/gtsam/geometry/geometry.i
@@ -856,7 +856,7 @@ class Cal3_S2Stereo {
   gtsam::Matrix K() const;
   gtsam::Point2 principalPoint() const;
   double baseline() const;
-  Vector6 vector() const;
+  gtsam::Vector6 vector() const;
   gtsam::Matrix inverse() const;
 };
 

